### PR TITLE
[MAINTENANCE] Change CategoricalColumnDomainBuilder property name from "limit_mode" to "cardinality_limit_mode".

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* It was determined that better context would be given to users at the `context.assistants.<name>.run()` level if directives had  more specific names (since the user does not have the same level of context as the developer of a `DomainBuilder`).
* Additional cleanup (typehints and directives usage in `DataAssistantRunner`).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-878/GREAT-843
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def _get_schema_instance(cls) -> Schema:
             return cls.get_config_class().schema_instance
 
     @classmethod
-    def from_commented_map(cls, commented_map: CommentedMap):
+    def from_commented_map(cls, commented_map: CommentedMap):  # type: ignore[no-untyped-def]
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:97]: Why are we doing this? Don't we know the type of the return?

File: great_expectations/rule_based_profiler/config/base.py [hunk 2]
```diff
@@ -521,32 +521,34 @@ def __init__(
...
-
-    @classmethod
-    def get_schema_class(cls) -> Type["RuleBasedProfilerConfigSchema"]:  # noqa: F821
-        return RuleBasedProfilerConfigSchema
-
-    @classmethod
-    def from_commented_map(
-        cls, commented_map: CommentedMap
-    ) -> "RuleBasedProfilerConfig":
+    def from_commented_map(cls, commented_map: CommentedMap):  # type: ignore[no-untyped-def]
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/config/base.py:524]: Same note throughout - I think `# type: ignore` should be used very sparingly.

Comment (cdkini) [reply great_expectations/rule_based_profiler/config/base.py:524]: I'd prefer `Any` over this tbh.

File: great_expectations/rule_based_profiler/rule_based_profiler.py [hunk 3]
```diff
@@ -881,32 +881,24 @@ def _apply_runtime_environment(
...
+                        existing_property_value = getattr(
+                            rule.domain_builder, property_key.value
+                        )
+                        property_value = BaseRuleBasedProfiler._get_effective_domain_builder_property_value(
+                            dest_property_value=property_value,
+                            source_property_value=existing_property_value,
+                        )
+                        setattr(rule.domain_builder, property_key.value, property_value)
+                    except AttributeError:
+                        pass
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/rule_based_profiler.py:895]: I don't love unchecked `try/except`. At the very least, I think we'd want a `log.debug`. Thoughts?

**Review by cdkini — CHANGES_REQUESTED** [review]
Changes requested around type hint directives and unchecked `try/except`

File: great_expectations/data_context/types/base.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/data_context/types/base.py:97]: @cdkini this was from before -- I just added the `ignore` macro.

File: great_expectations/rule_based_profiler/config/base.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/config/base.py:524]: @cdkini The `Any` here is not a great fit either.  Let us discuss what we can do.

**Review by anthonyburdi — COMMENTED** [review]
Looks good pending resolution of Chetan's comments!

**Review by cdkini — APPROVED** [review]

**Pull request merged**